core/aws-core/src/main/java/software/amazon/awssdk/awscore/rules/AwsEndpointProviderUtils.java

```diff
@@ -154,6 +160,91 @@ public static SdkHttpRequest setUri(SdkHttpRequest request, URI clientEndpoint,
                       .build();
     }
 
+    public static AwsRequest addHeaders(AwsRequest request, Map<String, List<String>> headers) {
```

Comment:
This method doesn't seem to belong to an endpoint utils class, it's not endpoint specific. Can we come up with a different place to put it?

Reply:
Meh, IMO t's a utils class *for* the providers to use. I'd rather just keep it here for now and move it if we need to to repeat this logic in other places

Reply:
I disagree in principle but I can agree to be pragmatic in this instance. It is probably easier to find a good structure when there is another usecase.